test/integration/bundle/cluster_bundle_validation_test.go

```diff
@@ -0,0 +1,275 @@
+			)
+
+			It("should require unique keys", func() {
+				selectorAccessor(&trustmanagerapi.KeyValueTarget{
+					Data: []trustmanagerapi.TargetKeyValue{{
+						Key: "foo",
+					}, {
+						Key: "foo",
+					}},
+				})
+				matchErr := "spec.target.%s.data[1]: Duplicate value: map[string]interface {}{\"key\":\"foo\"}"
+				Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field)))
+
+				selectorAccessor(&trustmanagerapi.KeyValueTarget{
+					Data: []trustmanagerapi.TargetKeyValue{{
+						Key: "foo",
+					}, {
+						Key: "bar",
+					}},
+				})
+				Expect(cl.Create(ctx, bundle)).To(Succeed())
+			})
+
+			DescribeTable("should prevent metadata with forbidden prefixes",
+				func(metadata *trustmanagerapi.TargetMetadata, wantErr bool) {
+					selectorAccessor(&trustmanagerapi.KeyValueTarget{
+						Metadata: metadata,
+						Data: []trustmanagerapi.TargetKeyValue{{
+							Key: "ca-bundle.crt",
+						}},
+					})
+					if wantErr {
+						var metadataField = "annotations"
+						if metadata.Labels != nil {
+							metadataField = "labels"
+						}
+						matchErr := "spec.target.%s.metadata.%s: Forbidden: must not use forbidden domains as prefixes (e.g., trust-manager.io)"
+						Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field, metadataField)))
+					} else {
+						Expect(cl.Create(ctx, bundle)).To(Succeed())
+					}
+				},
+				Entry("when trust-manager.io annotations are used", &trustmanagerapi.TargetMetadata{Annotations: map[string]string{"trust-manager.io/hash": "test"}}, true),
+				Entry("when trust-manager.io labels are used", &trustmanagerapi.TargetMetadata{Labels: map[string]string{"trust-manager.io/bundle": "bundle"}}, true),
+				Entry("when non-reserved annotations are used", &trustmanagerapi.TargetMetadata{Annotations: map[string]string{"not-trust-manager.io/hash": "test"}}, false),
+				Entry("when non-reserved labels are used", &trustmanagerapi.TargetMetadata{Labels: map[string]string{"not-trust-manager.io/bundle": "bundle"}}, false),
+			)
+
+			DescribeTable("should validate format",
+				func(format trustmanagerapi.BundleFormat, matchErr string) {
+					selectorAccessor(&trustmanagerapi.KeyValueTarget{
+						Data: []trustmanagerapi.TargetKeyValue{{
+							Key:    "ca-bundle",
+							Format: format,
+						}},
+					})
+					if matchErr != "" {
+						Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field)))
+					} else {
+						Expect(cl.Create(ctx, bundle)).To(Succeed())
+					}
+				},
+				Entry("when unset", trustmanagerapi.BundleFormat(""), ""),
+				Entry("when PEM", trustmanagerapi.BundleFormatPEM, ""),
+				Entry("when PKCS12", trustmanagerapi.BundleFormatPKCS12, ""),
+				Entry("when unknown", trustmanagerapi.BundleFormat("JKS"), "spec.target.%s.data[0].format: Unsupported value: \"JKS\": supported values: \"PEM\", \"PKCS12\""),
```

Comment:
question:

1) I presume the validation would reject "pem" or "PeM"? If so, is there any way to allow users to specify a value in a case-insensitive way? 

2) Either way, can we add a test for a bundle format with a different case just to capture what the behaviour is?